commands.js

```diff
@@ -1044,16 +1044,56 @@ exports.commands = {
 			move = Commands.randommove.call(this, "2,viable", self, dummyRoom);
 		}
 
+		// Code for generating a random name based on types generated
+		// Start by accessing the array from the namelist.js file in the data folder.
+		var foundName = false;
+		var t1Name = '';
+		var type1array = Tools.shuffle(Data.Namelist[type[0]]);
+
+		var type2array = [];
+		if (type[1] === undefined) { //Checking for single type case
```

Comment:
Though this works, it's kinda a weird way to do this check. `if (type.length === 1) {` does a better job describing what's being checked.